spring-pulsar/src/main/java/org/springframework/pulsar/core/DefaultPulsarConsumerFactory.java

```diff
@@ -89,9 +86,7 @@ public Consumer<T> createConsumer(Schema<T> schema, @Nullable Collection<String>
 		if (!CollectionUtils.isEmpty(customizers)) {
 			customizers.forEach(customizer -> customizer.customize(consumerBuilder));
 		}
-		Consumer<T> consumer = consumerBuilder.subscribe();
-		this.consumers.add(consumer);
```

Comment:
**[UNRELATED]** but a memory leak waiting to happen. I removed it as it was not being used by anything.